x-pack/test/observability_functional/apps/observability/pages/alerts/custom_threshold.ts

```diff
@@ -0,0 +1,254 @@
+        name: DATA_VIEW_2_NAME,
+        id: DATA_VIEW_2_ID,
+        title: DATA_VIEW_2,
+        logger,
+      });
+      await observability.alerts.common.navigateToRulesPage();
+    });
+
+    after(async () => {
+      await esArchiver.unload('x-pack/test/functional/es_archives/infra/metrics_and_logs');
+      // This also deletes the created data views
+      await kibanaServer.savedObjects.cleanStandardList();
+    });
+
+    it('shows the custom threshold rule in the observability section', async () => {
+      await observability.alerts.rulesPage.clickCreateRuleButton();
+      await observability.alerts.rulesPage.clickOnObservabilityCategory();
+      await observability.alerts.rulesPage.clickOnCustomThresholdRule();
+    });
+
+    it('can add name and tags', async () => {
+      await testSubjects.setValue('ruleNameInput', 'test custom threshold rule');
+      await testSubjects.setValue('comboBoxSearchInput', 'tag1');
+    });
+
+    it('can add data view', async () => {
+      // select data view
+      await testSubjects.click('selectDataViewExpression');
+      await testSubjects.setValue('indexPattern-switcher--input', 'test-data-view-name_2');
+      const dataViewExpression = await find.byCssSelector(
+        '[data-test-subj="indexPattern-switcher--input"]'
+      );
+      await dataViewExpression.pressKeys(Key.ENTER);
+      await retry.waitFor('data view selection to happen', async () => {
+        const dataViewSelector = await testSubjects.find('selectDataViewExpression');
+        return (await dataViewSelector.getVisibleText()) === 'DATA VIEW\ntest-data-view-name_2';
+      });
+    });
+
+    it('can select aggregation', async () => {
+      // select aggregation
+      await testSubjects.click('aggregationNameA');
+      await testSubjects.click('aggregationTypeSelect');
+      // assert all options are available
+      await find.byCssSelector('option[value="avg"]');
+      await find.byCssSelector('option[value="min"]');
+      await find.byCssSelector('option[value="max"]');
+      await find.byCssSelector('option[value="sum"]');
+      await find.byCssSelector('option[value="count"]');
+      await find.byCssSelector('option[value="cardinality"]');
+      await find.byCssSelector('option[value="p99"]');
+      await find.byCssSelector('option[value="p95"]');
+      await find.byCssSelector('option[value="rate"]');
+
+      // set first aggregation
+      await find.clickByCssSelector(`option[value="avg"]`);
+      const input1 = await find.byCssSelector('[data-test-subj="aggregationField"] input');
+      await input1.type('metricset.rtt');
+      await testSubjects.click('o11yClosablePopoverTitleButton');
+      await retry.waitFor('first aggregation to happen', async () => {
+        const aggregationNameA = await testSubjects.find('aggregationNameA');
+        return (await aggregationNameA.getVisibleText()) === 'AVERAGE\nmetricset.rtt';
+      });
+      await new Promise((r) => setTimeout(r, 1000));
```

Comment:
For future reference, we added these timeouts to ensure the selected data is passed to the alerting framework and the new rule params are available in the rule flyout before changing the next one; otherwise, an old state will be passed when updating the next rule param.